src/main/java/frc/robot/commands/PivotToClimbCommand.java

```diff
@@ -4,41 +4,75 @@
 
 package frc.robot.commands;
 
+import java.lang.Math;
 import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.DriveSubsystem;
 import frc.robot.subsystems.PivotAngleSubsystem;
+import frc.robot.Constants.PivotConstants;
 
 public class PivotToClimbCommand extends Command {
     private PivotAngleSubsystem pivotAngle;
+    private DriveSubsystem Drive;
     private double m_angle;
+    private double StartPosX;
+    private double StartPosY;
+    private double Distance = 0.0;
+    private boolean Driving = false;
 
     /** Creates a new PivotToClimbCommand. */
-    public PivotToClimbCommand(PivotAngleSubsystem pivotAngle, double angle) {
+    public PivotToClimbCommand(PivotAngleSubsystem pivotAngle, DriveSubsystem Drive, double angle) {
         this.pivotAngle = pivotAngle;
+        this.Drive = Drive;
         m_angle = angle;
         addRequirements(pivotAngle);
+        addRequirements(Drive);
         // Use addRequirements() here to declare subsystem dependencies.
     }
 
     // Called when the command is initially scheduled.
     @Override
     public void initialize() {
+        StartPosX = Drive.getPose().getX();
+        StartPosY = Drive.getPose().getY();
+        pivotAngle.setDesiredAngle(PivotConstants.kPivotTrapFirstPosition);
+        Driving = false;
+        Distance = 0;
+
     }
 
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
-        pivotAngle.setDesiredAngle(m_angle);
+        if (pivotAngle.getPitch() <= PivotConstants.kPivotTrapFirstPosition + 4.0) {
+            if (Driving == false) {
+                Drive.drive(PivotConstants.kPivotTrapDriveSPeed, 0.0, 0.0, false, true);
+                Driving = true;
+            }
+            double x = Drive.getPose().getX() - StartPosX;
+            double y = Drive.getPose().getY() - StartPosY;
+            double newAngle;
+            Distance = Math.sqrt(x * x + y * y);
+            newAngle = m_angle + (PivotConstants.kPivotTrapFirstPosition - m_angle)
+                    * (1 - Distance / PivotConstants.kPivotTrapDriveDistance);
+            if (newAngle > PivotConstants.kPivotTravelPosition)
+                newAngle = PivotConstants.kPivotTravelPosition;
+            if (newAngle < m_angle)
+                newAngle = m_angle;
+            pivotAngle.setDesiredAngle(newAngle);
```

Comment:
This is a really cool way of doing this! Having it raise based on how far it's driven is clever.